projects/vanilla/src/app/configurators/app-configuration/global-configurator.component.ts

```diff
@@ -79,11 +79,80 @@ import { ConfiguratorContext } from "@sinequa/ngx-ui-builder";
 
   <hr/>
 
+  <h6>Search Layout</h6>
+
+    <div class="form-check">
+      <input class="form-check-input" type="checkbox" id="layout-fullWidth" [(ngModel)]="context.config.layout.fullWidth" (ngModelChangeDebounced)="context.configChanged()">
+      <label class="form-check-label" for="layout-fullWidth">Full width</label>
+    </div>
+
+    <label class="form-label">Facets tab</label>
+    <div class="row mb-2">
+      <div class="col-3">
+        <label for="facets-sm" class="form-label">col-sm</label>
+        <input type="number" id="facets-sm" class="form-control" [(ngModel)]="context.config.layout.facets.sm" (ngModelChangeDebounced)="context.configChanged()">
+      </div>
+      <div class="col-3">
+        <label for="facets-md" class="form-label">col-md</label>
+        <input type="number" id="facets-md" class="form-control" [(ngModel)]="context.config.layout.facets.md" (ngModelChangeDebounced)="context.configChanged()">
+      </div>
+      <div class="col-3">
+        <label for="facets-lg" class="form-label">col-lg</label>
+        <input type="number" id="facets-lg" class="form-control" [(ngModel)]="context.config.layout.facets.lg" (ngModelChangeDebounced)="context.configChanged()">
+      </div>
+      <div class="col-3">
+        <label for="facets-xl" class="form-label">col-xl</label>
+        <input type="number" id="facets-xl" class="form-control" [(ngModel)]="context.config.layout.facets.xl" (ngModelChangeDebounced)="context.configChanged()">
+      </div>
+    </div>
+
+    <label class="form-label">Results tab</label>
+    <div class="row mb-2">
+      <div class="col-3">
+        <label for="results-sm" class="form-label">col-sm</label>
+        <input type="number" id="results-sm" class="form-control" [(ngModel)]="context.config.layout.results.sm" (ngModelChangeDebounced)="context.configChanged()">
```

Comment:
can you add some validation attributes to the inputs (min="0" max="12" step="1"). Also, for a value of 0, it won't work because "col-xx-0" is not defined in bootstrap. But maybe you could handle this special case by inserting a class d-xx-none?